### PR TITLE
Add authentication

### PR DESCRIPTION
**Pull request by MatthewStanciu**
Introducing: OAuth authentication in Next.js using only API Routes (no custom express server). Horrendous code, but it works. Will clean up later

### CODE REVIEW

File: pages/api/spotify.js [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
+const Spotify = require('spotify-web-api-node')
 const spotify = new Spotify()
 spotify.setRefreshToken(process.env.REFRESH_TOKEN)
 spotify.setClientId(process.env.CLIENT_ID)
 spotify.setClientSecret(process.env.CLIENT_SECRET)
 
-const refreshToken = () =>
-  new Promise((res, rej) => {
+const refreshToken = async () => {
+  return new Promise((res, rej) => {
```
Comment (lachlanjc) [inline pages/api/spotify.js:12]: This is a longer version of the same syntax, the return word & braces can be removed

File: pages/auth.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      (req ? `http://${req.headers.host}` : '') + '/api/authdetails'
+    )
+    const data = await authData.json()
+
+    const code = req.url.match('=(.*)')[1]
+    data.code = code
+
+    const getRefreshToken = async () => {
+      return new Promise((resolve, reject) => {
+        axios({
```
Comment (lachlanjc) [inline pages/auth.js:17]: Why are we using 3 different libraries for the exact same thing :)

File: pages/auth.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          method: 'POST',
+          url: `https://accounts.spotify.com/api/token`,
+          params: {
+            grant_type: 'authorization_code',
+            code: code,
+            redirect_uri: 'http://localhost:3000/auth'
```
Comment (lachlanjc) [inline pages/auth.js:23]: Should use current origin, right?

File: pages/auth.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ refresh_token: refreshToken })
+    })
+      .then(r => console.log('success'))
+      .catch(err => console.log(err))
+
+    res.writeHead(302, {
+      Location: `http://localhost:3000/app`
```
Comment (lachlanjc) [inline pages/auth.js:56]: Again, match the origin

File: pages/auth.js (hunk 2 above)
Comment (MatthewStanciu) [reply pages/auth.js:17]: becoz i wanna,

**Pull request merged**